server/util.go

```diff
@@ -29,21 +29,33 @@ import (
 
 // server context
 type Context struct {
-	Config *cfg.Config
-	Logger log.Logger
+	Config          *cfg.Config
+	Logger          log.Logger
+	DefaultNodeHome string
+	DefaultCLIHome  string
+}
+
+// NewContext creates a new Context object
+func NewContext(config *cfg.Config, logger log.Logger,
+	defaultNodeHome string, defaultCLIHome string) *Context {
+
+	return &Context{
+		Config:          config,
+		Logger:          logger,
+		DefaultNodeHome: defaultNodeHome,
+		DefaultCLIHome:  defaultCLIHome,
```

Comment:
I think this is not the right place where to hold this information. Why should the server know where the client lives after all?